File: Python Course/Software Development Module - in Python/Week3/EggsApp.py
"""
EggsApp.py
Billy Cussen
25/09/2020
"""
#Objects
import gc
import Eggs
from Eggs import myEggs

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = True

    while inputCheck:
        try:
            eggs = int(input("Please enter the number of eggs collected today: "))
        except:
            print("You need to enter an Integer (e.g. 1, 2)")
        else:
            if eggs < 0:
                print("You need to enter a number of 0 or greater")
            else:
                inputCheck = False

    eggsApp = myEggs(eggs)
    eggsApp.computeBoxesAndRemainder()
    print("Number of Boxes: "+str(eggsApp.boxes))
    print("Remaining Eggs: "+str(eggsApp.remainder))
    #Run GC and free memory
    eggsApp = None
    gc.collect()

    repeat = ""

    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to repeat(\"Y\" or \"N\"): "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print("You need to enter Y or N")
        
print("Thanks for using my App")

File: Java/Software Development Module - Java/Week3/EggsApp.java
/*
EggsApp.java
Billy Cussen
08/07/2020
*/

//This is the App Class for Eggs.java. This is the executable class

import java.util.InputMismatchException;
import java.util.Scanner;

public class EggsApp {

    public static void main(String [] args){

        //Variables
        int eggs = 0, remainder, boxes;
        boolean checkInputEggs;

        //Objects
        Scanner keyboard;
        Eggs myEggs = new Eggs();

        //Input
        do{
            keyboard = new Scanner(System.in);
            checkInputEggs = true;

            try{
                System.out.print("Please enter collected eggs for today: ");
                eggs = keyboard.nextInt();
            } catch(InputMismatchException e){
                System.out.println("You need to enter eggs collected as an Integer (e.g. 24)");
                checkInputEggs = false;
            }

        }while(!checkInputEggs);

        //Setter
        myEggs.setEggs(eggs);

        //Compute
        myEggs.computeBoxes();

        //Output
        boxes = myEggs.getBoxes();
        remainder = myEggs.getRemainder();

        System.out.println("You collected "+eggs+" today.");
        System.out.println("There are "+boxes+" boxes, with "+remainder+" eggs remaining");

    }
}